contracts/adapters/FixedRebalanceExtension.sol

```diff
@@ -110,24 +112,24 @@ contract FixedRebalanceExtension is BaseExtension {
      * @param _maturities                Relative maturities (i.e. "3 months") in seconds
      * @param _allocations               Relative allocations (i.e. 0.9 = 90%) with 18 decimals corresponding to the respective maturity
      */
-    function setAllocations(uint256[] memory _maturities, uint256[] memory _allocations) external onlyOperator {
-        _setAllocations(_maturities, _allocations);
+    function setAllocations(uint256[] memory _maturities, uint256[] memory _allocations, uint256[] memory _minPositions) external onlyOperator {
+        _setAllocations(_maturities, _allocations, _minPositions);
     }
 
     /**
-     * ONLY OPERATOR: Rebalances the positions towards the configured allocation percentages.
+     * ONLY ALLOWED CALLER: Rebalances the positions towards the configured allocation percentages.
      *
-     * @param _share                Relative share of the necessary trade volume to execute (allows for splitting the rebalance over multiple transactions
-     * @param _minPositions         Minimum positions (in set token units) for each maturity after rebalance. (slippage protection)
+     * @param _share                 Relative share of the necessary trade volume to execute (allows for splitting the rebalance over multiple transactions
+     * @param _rebalanceMinPositions Minimum positions (in set token units) for each maturity after this rebalance operation. 
+     * @dev Will revert if _rebalanceMinPositions is lower than the minPositions configured by the operator (weighted by share)
      */
-    function rebalance(uint256 _share, uint256[] memory _minPositions) external onlyOperator {
+    function rebalance(uint256 _share, uint256[] memory _rebalanceMinPositions) external onlyAllowedCaller(msg.sender) {
```

Comment:
todo: return the positions to the caller so that this function can be simulated to find optimal _rebalanceMinPositions

Reply:
Done